src/Storefront/Resources/views/components/README.md

```diff
@@ -34,6 +34,122 @@ components/Button/
 <twig:sw-button>My button</twig:sw-button>
 ```
 
+## Using a PHP class for the component
+
+Besides the anonymous components, which only require a Twig template, you can also define your component via a PHP class which offers the possiblity to add additional businsess logic to your component. You can have a look at the [offical documentation](https://symfony.com/bundles/ux-twig-component/current/index.html) for all the details.
+
+In Shopware we decided that these PHP classes should be placed right where your component template and other files of your component are located. This provides the epxerience of a real comopnent system and you have all componente related files in one place. Therefore you can simply add the PHP class to the described directory structure.
+
+```
+views/components/Button/
+    Button.html.twig
+    Button.js
+    Button.php
+    Button.scss
+```
+
+The loading and template matching is already solved by placing the file in the right directory, so you don't have to define a secific name or template path in your component class.
+
+```PHP
+<?php declare(strict_types=1);
+
+namespace Shopware\Storefront\Resources\views\components\Button;
+
+use Symfony\UX\TwigComponent\Attribute\AsTwigComponent;
+
+#[AsTwigComponent()]
+class Button
+{
+    public string $text = 'Click me!';
+}
+```
+
+The only thing you have to do though is to register your component class as a service in the service.xml configuration.
+
+```XML
+<!-- services.xml -->
+
+<!-- other config ... -->
+
+<service id="Shopware\Storefront\Resources\views\components\Button">
+    <tag name="twig.component" />
+</service>
+```
+
+Now your component is all set up and ready to go. You can directly access it in the template under the given name and namespace.
+
+**Note:** Be carefull with adding too much business logic inside component classes. At Shopware we mostly implement anonymous components that are pure frontend representaions which get passed all necessary data via properties. 
+
+
+## Adding components via Shopware extension
+
+New components can easiley be added by Shopware apps and plugins in the same way as described above.
+
+### Anonymous Components
+The anonymous components work right of the box and you dont't have to do anything else than putting your component files under `<your-plugin>/src/Resources/views/components`. Both, apps and plugins can provide anonymous components in any distribution form of Shopware. Your components will automatically be prefixed with the bundle name of your app or plugin. 
```

Comment:
Maybe specify that by component files, we mean that only a twig template file is enough.